benchmarks/distributed/test_many_tasks.py

```diff
@@ -1,47 +1,111 @@
+import asyncio
 import click
 import json
 import os
 import ray
-import ray._private.test_utils as test_utils
 import time
 import tqdm
+import numpy as np
 
-sleep_time = 300
+
+@ray.remote(num_cpus=0)
+class SignalActor:
+    def __init__(self):
+        self.timestamps = []
+
+    def reset(self, num_expected):
+        self.start_time = time.time()
+        self.num_expected = num_expected
+
+    def send(self):
+        self.timestamps.append(time.time() - self.start_time)
+        assert len(self.timestamps) <= self.num_expected
+
+    def get_timestamps(self):
+        return self.timestamps
+
+
+@ray.remote
+def task(i, signal, semaphore):
+    signal.send.remote()
+    ref = semaphore.acquire.remote()
+    ready = False
+    while not ready:
+        ready, _ = ray.wait([ref], timeout=0)
+        time.sleep(1)
+
+
+@ray.remote(num_cpus=0)
+class Semaphore:
```

Comment:
nit: should we just use `test_utils.Semaphore`?

Reply:
I prefer doing it this way to keep the test independent of private APIs.